agent/controller/controller.go

```diff
@@ -0,0 +1,92 @@
+/* Copyright 2017 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package controller contains methods that related to AP controller.
+package controller
+
+import (
+	"errors"
+	"time"
+
+	devctx "github.com/google/link022/agent/context"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+
+	log "github.com/golang/glog"
+	pb "github.com/openconfig/gnmi/proto/gnmi"
+)
+
+const (
+	hostnameHeader       = "hostname"
+	gnmiAddrHeader       = "gnmi_address"
+	resyncRequiredHeader = "resync_required"
+
+	heartbeatInterval = 15 * time.Second
+)
+
+// ReportAPInfo reports the current AP information to the assigned controller.
+// It returns error if failures detected.
+func ReportAPInfo(syncRequried bool) error {
```

Comment:
Can this be a private func?

Reply:
This will be used in other package when AP device requests to sync with the latest config from controller. Plan to do that in later CLs.